### ISSUE TITLE
Implement Pooling of API requests

### ISSUE THREAD

**Issue by Demetrio92** [rodazuero/gmapsdistance#44]
The way we call gmaps API right now can be optimised by stitching together up to 25 request using `paste(origins, collapse = "|")`/`paste(destinations, collapse = "|")`. 

Documentation by google is here: https://developers.google.com/maps/documentation/distance-matrix/intro

**Comment by Demetrio92** [comment]
Created a new issue based on #12

**Comment by MichaelChirico** [comment]
I was looking into working on this & noticed some changes to API billing... is it correct that there's no longer _any_ free quota for requests?

**Comment by rodazuero** [comment]
@MichaelChirico yes, it seems like there are no longer free quotas at all. You need to create an account...
